pkg/services/hivelocity/device/device.go

```diff
@@ -154,7 +154,21 @@ func GetFirstDevice(ctx context.Context, hvclient hvclient.Client, machineType i
 		return devices[i].DeviceId < devices[j].DeviceId
 	})
 
-	return findAvailableDeviceFromList(devices, machineType, clusterName, machineName)
+	hvDevices, err := findAvailableDeviceFromList(devices, machineType, hvCluster.Name, machineName)
+	if err != nil {
+		if errors.Is(err, errNoDeviceAvailable) {
+			conditions.MarkFalse(
```

Comment:
we shouldn't return an error here. however, this would mean that our hv.BareMetaldevice can be nil, just as the error. 

Let's talk about it if you don't know how to solve it. How is it in CAPH BM?